### ISSUE TITLE
Cannot upgrade chart to use persistent volume claim

### ISSUE THREAD

**Issue by domcar** [coreos/prometheus-operator#1045]
**What did you do?**
helm upgrade --install --kube-context default --namespace aux
**What did you expect to see?**
Chart upgraded
**What did you see instead? Under which circumstances?**
Error: UPGRADE FAILED: failed to create patch: unable to find api field in struct Unstructured for the json field "spec"
**Environment**
GKE 1.9
* Kubernetes version information:

	Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.6", GitCommit:"6260bb08c46c31eea6cb538b34a9ceb3e406689c", GitTreeState:"clean", BuildDate:"2017-12-21T06:34:11Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.3-gke.0", GitCommit:"a7b719f7d3463eb5431cf8a3caf5d485827b4210", GitTreeState:"clean", BuildDate:"2018-02-16T18:26:01Z", GoVersion:"go1.9.2b4", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

	Terraform.

After deploying kube-prometheus (helm chart) I decided to upgrade the chart to use the persistent volume. This is what I change in the value file:
```
-  storageSpec: {}
-  #  volumeClaimTemplate:
-  #    spec:
-  #      resources:
-  #        requests:
-  #          storage: 50Gi
+  storageSpec:
+    volumeClaimTemplate:
+      spec:
+        resources:
+          requests:
+            storage: 50Gi
```
Basically I just uncommented the lines.
But when I tried to upgrade I receive the error I posted above. I tried this on 2 different clusters with the same result. 
If instead I delete kube-prometheus chart and reinstall with the volumeClaimTemplate enabled it works.

**Comment by gianrubio** [comment]
@domcar I did a PR #1074 to fix the `storageSpec`, somehow this document was not updated since the last change in `storageSpec`. Please adapt your values to the same format suggested in the new example.
